File: backend/routes/onboarding_routes.py
"""Onboarding routes."""
from __future__ import annotations
from fastapi import APIRouter, Depends

from middleware.auth_middleware import require_auth
from services import onboarding_service

router = APIRouter(prefix="/v1/users/me/onboarding-checklist", tags=["onboarding"])


@router.get("")
async def get_state(user=Depends(require_auth)):
    # Also attempt to grant the bonus lazily if all steps complete
    return await onboarding_service.maybe_grant_bonus(user.id)

File: backend/src/routes/onboarding.routes.js
const express = require('express');
const { requireAuth } = require('../middleware/auth.middleware');
const onboardingService = require('../services/onboarding.service');
const { catchAsync } = require('../utils/helpers');

const router = express.Router();

router.get('/', requireAuth, catchAsync(async (req, res) => {
  const result = await onboardingService.maybeGrantBonus(req.user._id.toString());
  res.json(result);
}));

module.exports = router;
